neuro_san/http_sidecar/handlers/base_request_handler.py

```diff
@@ -124,6 +131,28 @@ def get_concierge_grpc_session(self, metadata: Dict[str, Any]) -> ConciergeSessi
                 metadata=metadata)
         return grpc_session
 
+    async def update_agents(self, metadata: Dict[str, Any]) -> bool:
+        """
+        Update internal agents table by executing request
+        to underlying gRPC service.
+        :param metadata: metadata for request from caller context.
+        :return: True if update was successful
+                 False otherwise
+        """
```

Comment:
Execute "concierge" gRPC request to the neuro-san host to get the current list of known (allowed) agents.